src/ledger/ledger-radix-babylon.ts

```diff
@@ -0,0 +1,87 @@
+import Transport from '@ledgerhq/hw-transport'
+import { errAsync, ok, ResultAsync } from 'neverthrow'
+import { LedgerPublicKeyRequest } from './schemas'
+
+export const encodeBip32 = (hdPath: string) => {
+  const path = hdPath.slice(2).split('/')
+  const length = `00${(path.length & 0xff).toString(16)}`.slice(-2)
+
+  const parts = path
+    .map((value) => 0x80000000 + parseInt(value, 10))
+    .map((value) => value.toString(16))
+    .join('')
+
+  return `${length}${parts}`
+}
+
+export enum LedgerErrorResponse {
+  UnlockDevice = '5515',
+  OpenRadixApp = 28161,
+}
+
+export enum LedgerInstructionCode {
+  GetDeviceModel = 0x11,
+  GetDeviceId = 0x12,
+  GetPubKeyEd25519 = 0x21,
+  GetPubKeySecp256k1 = 0x31,
+}
+
+export const errorResponses: Record<string, string> = {
+  [LedgerErrorResponse.UnlockDevice]:
+    'Please unlock Ledger Device and try again',
+  [LedgerErrorResponse.OpenRadixApp]:
+    'Please open Radix Babylon app in your Ledger device and try again',
+}
+
+export class LedgerRadixBabylon {
```

Comment:
Is there a specific reason why you are using a class? Same functionality can be accomplished with a function and not introduce classes into the codebase

Reply:
I can't find the article but all ledger-live apps like [BTC](https://github.com/LedgerHQ/ledger-live/tree/develop/libs/ledgerjs/packages/hw-app-btc) and [ETH](https://github.com/LedgerHQ/ledger-live/tree/develop/libs/ledgerjs/packages/hw-app-eth) implement same interface I wanted to follow. Later I realized we don't need it so moving back to function and closure for now :)